modules/tftp/server.rb

```diff
@@ -90,21 +103,114 @@ def pxe_default
     end
 
     def pxeconfig_file(mac)
-      ["#{pxeconfig_dir}/menu.lst.01" + mac.delete(':').upcase, "#{pxeconfig_dir}/01-" + mac.tr(':', '-').upcase]
+      ["#{pxeconfig_dir}/menu.lst.01" + mac.delete(':').upcase, "#{pxeconfig_dir}/01-" + dashed_mac(mac).upcase]
     end
   end
 
   class Pxegrub2 < Server
-    def pxeconfig_dir
-      "#{path}/grub2"
+    def bootloader_path(os, release, arch)
+      bootloader_path = File.join(Proxy::TFTP::Plugin.settings.tftproot, 'bootloader-universe/pxegrub2', os, release, arch)
+
+      logger.debug "TFTP: Checking if bootloader universe is configured for OS '#{os}' version '#{release}' (#{arch})."
+      logger.debug "TFTP: Checking if directory '#{bootloader_path}' exists."
+      unless Dir.exist?(bootloader_path)
+        logger.debug "TFTP: Directory '#{bootloader_path}' does not exist."
+
+        bootloader_path = File.join(Proxy::TFTP::Plugin.settings.tftproot, 'bootloader-universe/pxegrub2', os, 'default', arch)
+
+        logger.debug "TFTP: Checking if bootloader universe is configured for OS '#{os}' (#{arch})."
+        logger.debug "TFTP: Checking if directory '#{bootloader_path}' exists."
+        unless Dir.exist?(bootloader_path)
+          logger.debug "TFTP: Directory '#{bootloader_path}' does not exist."
+          return
+        end
+      end
+
+      bootloader_path
+    end
+
+    def create_bootloader_universe_symlinks(bootloader_path, pxeconfig_dir_mac)
+      symlinks = []
+      Dir.glob(File.join(bootloader_path, '*.efi')).each do |source_file|
+        symlinks << { source: Pathname.new(source_file), symlink: Pathname.new(File.join(pxeconfig_dir_mac, File.basename(source_file))) }
```

Comment:
Could you explain the choice of using `Pathname` here? Is there a reason we couldn't simplify this by using `File.relative_path` instead?

Reply:
Sorry... I wrote about that in a [comment above](https://github.com/theforeman/smart-proxy/pull/877?notification_referrer_id=NT_kwDOAkJiPbM4MDY3MzgzMjE5OjM3OTA0OTU3#discussion_r1732772664) but maybe I should have made it more visible.

I saw that the `File.relative_path` method comes from the `yard` RubyGem and as far as I know this is not a dependency of Smart Proxy. Therefore I searched for a different approach. Please let me know if I missed something and my assumption is wrong. :)